lib/modules/apostrophe-login/index.js

```diff
@@ -266,19 +266,19 @@ module.exports = {
         user._permissions.guest = true;
       }
 
-      // TODO: make this a deprecated default behavior (sigh)
-
       // If you are admin- for any type of content, you need to be
       // at least guest to effectively attach media to your content,
       // and the edit permission also makes sense because it does not
       // immediately let you do anything, just makes it easier to see
       // you are a candidate to do things like edit pages if given
       // specific rights to them. Also simplifies outerLayout's logic
-      if (_.some(user._permissions, function(val, key) {
-        return key.match(/^admin-/);
-      })) {
-        user._permissions.guest = true;
-        user._permissions.edit = true;
+      if (!self.apos.permissions.options.typeAdminsCanEditAllTypes) {
```

Comment:
This changes the default behavior, which is a bc break. Should read:

```
if (self.apos.permissions.options.typeAdminsCanEditAllTypes !== 'false')
```

So it must be explicitly turned off.